### PR TITLE
qtile binary: don't fail if we can't set the locale

### LINKED ISSUE #1577: qtile error to load
in load qtile ```Traceback (most recent call last):
  File "/usr/bin/qtile", line 11, in <module>
    load_entry_point('qtile==0.14.2', 'console_scripts', 'qtile')()
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 490, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2859, in load_entry_point
    return ep.load()
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2450, in load
    return self.resolve()
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2456, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/lib/python3.8/site-packages/libqtile/scripts/qtile.py", line 32, in <module>
    locale.setlocale(locale.LC_ALL, locale.getdefaultlocale())  # type: ignore
  File "/usr/lib/python3.8/locale.py", line 608, in setlocale
    return _setlocale(category, locale)
locale.Error: unsupported locale setting```

my locale settings ```
locale: Cannot set LC_CTYPE to default locale: No such file or directory
locale: Cannot set LC_MESSAGES to default locale: No such file or directory
locale: Cannot set LC_ALL to default locale: No such file or directory
LANG=es-US.UTF-8
LC_CTYPE="es-US.UTF-8"
LC_NUMERIC="es-US.UTF-8"
LC_TIME="es-US.UTF-8"
LC_COLLATE="es-US.UTF-8"
LC_MONETARY="es-US.UTF-8"
LC_MESSAGES="es-US.UTF-8"
LC_PAPER="es-US.UTF-8"
LC_NAME="es-US.UTF-8"
LC_ADDRESS="es-US.UTF-8"
LC_TELEPHONE="es-US.UTF-8"
LC_MEASUREMENT="es-US.UTF-8"
LC_IDENTIFICATION="es-US.UTF-8"
LC_ALL=```

### PR DESCRIPTION
**Pull request by tych0**
This isn't really fatal, so let's nto make it that way.

Fixes #1577

Signed-off-by: Tycho Andersen <tycho@tycho.ws>

### CODE REVIEW

File: libqtile/scripts/qtile.py [hunk 1]
```diff
@@ -29,7 +29,11 @@
 from libqtile import confreader
 from libqtile.backend.x11 import xcore
 
-locale.setlocale(locale.LC_ALL, locale.getdefaultlocale())  # type: ignore
+try:
+    locale.setlocale(locale.LC_ALL, locale.getdefaultlocale())  # type: ignore
+except locale.Error:
+    pass
```
Comment (ramnes) [inline libqtile/scripts/qtile.py:35]: Maybe log something?

Comment (tych0) [reply libqtile/scripts/qtile.py:35]: I thought about that, but we don't have the log set up here yet. By the time we do, some of the locale stuff may already be processed.

Comment (tych0) [reply libqtile/scripts/qtile.py:35]: We could figure it out if we need, but I think the only reason we use the locale right now is for currency in the bitcoin/stock ticker widget, so it's pretty low priority :)

Comment (arcolinuxz) [reply libqtile/scripts/qtile.py:35]: I have tried this in the live usb - does not seem to help

Comment (arcolinuxz) [reply libqtile/scripts/qtile.py:35]: Afterwards the last line seems to be different - confreader
error-code

**Comment by tych0** [comment]
On Thu, Apr 02, 2020 at 11:19:19AM -0700, arcolinuxz wrote:
> @arcolinuxz commented on this pull request.
> 
> 
> 
> > @@ -29,7 +29,11 @@
>  from libqtile import confreader
>  from libqtile.backend.x11 import xcore
>  
> -locale.setlocale(locale.LC_ALL, locale.getdefaultlocale())  # type: ignore
> +try:
> +    locale.setlocale(locale.LC_ALL, locale.getdefaultlocale())  # type: ignore
> +except locale.Error:
> +    pass
> 
> Afterwards the last line seems to be different - confreader
> error-code

So it does help, the error is just different. What version of python?
Seems like it doesn't understand type annotations.

**Comment by arcolinuxz** [comment]
Pacman says 3.8.2-1

**Comment by arcolinuxz** [comment]
I did see a folder on usr/lib with another version
the one you are using I think
python 2.7
and
python3.8

**Comment by tych0** [comment]
On Thu, Apr 02, 2020 at 11:24:58AM -0700, arcolinuxz wrote:
> I did see a folder on usr/lib with another version
> the one you are using I think
> python 2.7
> and
> python3.8

Qtile no longer supports python2, start it with python3.

**Comment by arcolinuxz** [comment]
Same thing - I think we are using python3
`python3 /usr/bin/qtile`
That is what you mean right?

**Comment by arcolinuxz** [comment]
Starting it with python2 gives an error too

**Comment by tych0** [comment]
On Thu, Apr 02, 2020 at 11:57:31AM -0700, arcolinuxz wrote:
> Same thing - I think we are using python3
> `python3 /usr/bin/qtile`
> That is what you mean right?

I can't see your whole traceback, specifically the bottom part. Can
you please also copy and paste them so we don't need to load a URL?

**Comment by arcolinuxz** [comment]
I have rebooted the liveusb.
back on qtile (not git)
start

**Comment by arcolinuxz** [comment]
Then the code change you suggested at the top
code-change

**Comment by arcolinuxz** [comment]
We have a successful launch of qtile

**Comment by arcolinuxz** [comment]
please put the code in the main branch

File: libqtile/scripts/qtile.py (hunk 1 above)
Comment (arcolinuxz) [reply libqtile/scripts/qtile.py:35]: I confirm this code would fix qtile crashing on ArcoLinux when used together on iso and lightdm.

Comment (arcolinuxz) [reply libqtile/scripts/qtile.py:35]: #1581 See trials and successful installation

**Pull request merged**